### PR TITLE
Add Field in the remotewrite CRD to sync secrets

### PR DESCRIPTION
**Pull request by mcbenjemaa**
Towards: https://github.com/giantswarm/roadmap/issues/1180
 
Add Secrets field to remotewrite CRD, in order to sync them to the Prometheus namespace

```yaml
secrets:
- name: grafana-cloud-basic-auth
  data:
    username: <Base64>
    password: <Base64> 
```

Ref: https://github.com/giantswarm/roadmap/issues/1180#issuecomment-1162802666

## Checklist

I have:

- [X] Described why this change is being introduced
- [ ] Separated out refactoring/reformatting in a dedicated PR
- [X] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

File: api/v1alpha1/remoteWrite_types.go [hunk 1]
```diff
@@ -17,6 +17,16 @@ type RemoteWrite struct {
 type RemoteWriteSpec struct {
 	RemoteWrite     pov1.RemoteWriteSpec `json:"remoteWrite"`
 	ClusterSelector metav1.LabelSelector `json:"clusterSelector"`
+
+	// Secrets that will be created in the relevant Prometheus namespace
+	// +optional
+	// +immutable
+	Secrets []RemoteWriteSecretSpec `json:"secrets,omitempty"`
```
[the remark this answers is not in the document]
Comment (mcbenjemaa) [reply api/v1alpha1/remoteWrite_types.go:24]: Unfortunately, the controller looks for secrets in Prometheus namespace, so we need to sync the secrets 
By adding this field, we will make sure the controller creates them in the underlying namespace.

File: api/v1alpha1/remoteWrite_types.go [hunk 2]
```diff
@@ -17,6 +17,16 @@ type RemoteWrite struct {
 type RemoteWriteSpec struct {
 	RemoteWrite     pov1.RemoteWriteSpec `json:"remoteWrite"`
 	ClusterSelector metav1.LabelSelector `json:"clusterSelector"`
+
+	// Secrets that will be created in the relevant Prometheus namespace
```
Comment (TheoBrigitte) [inline api/v1alpha1/remoteWrite_types.go:21]: I would change the description to something like

```suggestion
	// Secrets data to be created along with the configured Prometheus resource.
	// This provides the data for any v1.SecretKeySelector used in the subsequent RemoteWrite field.
	// Provided name and keys should match values in v1.SecretKeySelector fields.
```

I am also thinking we should provide an example somewhere to illustrate how this works.

Comment (mcbenjemaa) [reply api/v1alpha1/remoteWrite_types.go]: https://github.com/giantswarm/prometheus-meta-operator/pull/928/files#diff-a8f899db168f9bb204ce8254693b9911f9e06e693f5647e447a4df2b828b54f9R28

Comment (TheoBrigitte) [reply api/v1alpha1/remoteWrite_types.go]: Nice example :+1:

**Review by TheoBrigitte — APPROVED** [review]

**Pull request merged**